### PR TITLE
Implement user registration

### PR DESCRIPTION
**Pull request by dmytrostriletskyi**

### CODE REVIEW

File: src/components/pages/SignUpPage.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+      console.log(mutation.type)
+      console.log(authenticationStorageMutations.subscribe.markAsSignedUp)
+
+      if (mutation.type === authenticationStorageMutations.subscribe.addError) {
+        this.error = state.authentication.error
+      }
+
+      if (mutation.type === authenticationStorageMutations.subscribe.addFieldsErrors) {
+        this.fieldsErrors = state.settings.fieldsErrors
```
Comment (anastasiia-bilova) [inline src/components/pages/SignUpPage.vue]:
```suggestion
        this.fieldsErrors = state. authentication.fieldsErrors
```

File: src/components/pages/SignUpPage.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        email: this.email,
+        username: this.username,
+        password: this.password,
+      })
+    }
+  },
+  mounted() {
+    store.subscribe((mutation, state) => {
+
+      console.log(mutation.type)
```
Comment (anastasiia-bilova) [inline src/components/pages/SignUpPage.vue]: Remove debug code, please.

File: src/store/modules/authentication.js [hunk 3]
```diff
@@ -25,32 +37,62 @@ export const authentication = {
...
+          .post(`https://bps-directory-back-staging.herokuapp.com/users/registration/`, {
+            email: email,
+            username: username,
+            password: password,
+          })
+          .then(response => {
+            commit(authenticationStorageMutations.commit.markAsSignedUp)
+          })
+          .catch(error => {
+            console.log(error.response)
```
Comment (anastasiia-bilova) [inline src/store/modules/authentication.js]: Remove debug code.

**Review by anastasiia-bilova — APPROVED** [review]

File: src/components/pages/SignUpPage.vue (hunk 2 above)
Comment (dmytrostriletskyi) [reply src/components/pages/SignUpPage.vue]: @anastasiia-bilova, done.

File: src/components/pages/SignUpPage.vue (hunk 1 above)
Comment (dmytrostriletskyi) [reply src/components/pages/SignUpPage.vue]: @anastasiia-bilova, done.

File: src/store/modules/authentication.js (hunk 3 above)
Comment (dmytrostriletskyi) [reply src/store/modules/authentication.js]: @anastasiia-bilova, done.

**Pull request merged**